From 30fecfa188455a560aeea6f6dc60bc2f7c19bb14 Mon Sep 17 00:00:00 2001
From: James Ren <12345678goldlionren@users.noreply.github.com>
Date: Thu, 13 Aug 2026 02:43:45 +1000
Subject: [PATCH 2/2] Fix sampling control and Arabic normalization

---
 indextts/infer_v2_5.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/indextts/infer_v2_5.py b/indextts/infer_v2_5.py
index 0454cd9..447f74f 100644
--- a/indextts/infer_v2_5.py
+++ b/indextts/infer_v2_5.py
@@ -714,7 +714,7 @@ class IndexTTS2:
         if text_normalization:
             if lang.lower() in ['zh', 'zhen', 'en']:
                 text = self.text_process.normalize(text)
-            elif lang.lower() in ['ja', 'es']:
+            elif lang.lower() in ['ja', 'es', 'ar']:
                 text = nemo_text_normalize(text, lang.lower())
             print(f'text after normalization: {text}')
 
@@ -791,7 +791,7 @@ class IndexTTS2:
                         emo_vec=emovec,
                         campplus_embedding=style,
                         wav=spk_audio_prompt,
-                        do_sample=True,
+                        do_sample=do_sample,
                         top_p=top_p,
                         top_k=top_k,
                         temperature=temperature,
-- 
2.52.0.windows.1

